### PR TITLE
[K32W] Add Gn/Ninja for the ELock Demo App

### PR DESCRIPTION
**Pull request by doru91**
Problem
Autoconf/Automake is deprecated.

Summary of Changes
Add support for GN/Ninja Build System for the NXP K32W061 ELock Demo Application.

### CODE REVIEW

**Comment by doru91** [comment]
There is one test failing:

2s
Run github/codeql-action/analyze@v1
  with:
    output: ../results
    upload: true
    add-snippets: false
    checkout_path: /home/runner/work/connectedhomeip/connectedhomeip
    token: ***
    matrix: null
  env:
    BUILD_TYPE: nrfconnect
/usr/bin/docker exec  0b8ae3e04851628e8b6ba7f838f3788755083411cd975b12c17c233ddeacbd7f sh -c "cat /etc/*release | grep ^ID"
Error: Config file could not be found at expected location. Has the 'init' action been called?
Error: Config file could not be found at expected location. Has the 'init' action been called?
    at run (/__w/_actions/github/codeql-action/v1/lib/analyze-action.js:40:19)

Is this error cause by my PR?

**Comment by rwalker-apple** [comment]
Would you mind updating the description using the PR template?

**Comment by rwalker-apple** [comment]
@woody-apple the nrfconnect failure looks related to CodeQL

**Comment by doru91** [comment]
> Would you mind updating the description using the PR template?

Updated the description of the PR.

**Comment by rwalker-apple** [comment]
needs a rebase to pick up the CodeQL skip

**Comment by rwalker-apple** [comment]
> > Would you mind updating the description using the PR template?
> 
> Updated the description of the PR.

thank you!

**Review by BroderickCarlin — APPROVED** [review]

**Comment by doru91** [comment]
> needs a rebase to pick up the CodeQL skip

Done.

File: third_party/k32w_sdk/k32w_sdk.gni [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      "-Wno-conversion",
+    ]
+
+    # Now add our "system-header" include dirs
+    foreach(include_dir, _sdk_include_dirs) {
+      cflags += [ "-isystem" + rebase_path(include_dir, root_build_dir) ]
+    }
+  }
+
+  # TODO - Break up this monolith and make it configurable.
```
Comment (andy31415) [inline third_party/k32w_sdk/k32w_sdk.gni:138]: should we create a GH issue for this and reference it here instead of a simple TODO?

**Review by andy31415 — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

File: third_party/k32w_sdk/k32w_sdk.gni (hunk 1 above)
Comment (doru91) [reply third_party/k32w_sdk/k32w_sdk.gni:138]: Created https://github.com/project-chip/connectedhomeip/issues/3219. How do I create a reference to it?

**Pull request merged**